packages/api-client-core/src/FieldSelection.ts

```diff
@@ -3,5 +3,28 @@
  * Example: `{ id: true, name: false, richText: { markdown: true, html: false } }`
  **/
 export interface FieldSelection {
-  [key: string]: boolean | null | undefined | FieldSelection;
+  [key: string]: FieldSelectionType;
+}
+
+export type FieldSelectionType = boolean | null | undefined | FieldSelection | ConnectionFieldSelection;
+
+export interface ConnectionFieldSelection {
+  before?: string;
+  after?: string;
+  first?: number;
+  last?: number;
+  pageInfo?: FieldSelection;
+  edges: EdgeFieldSelection;
+}
+
+export interface EdgeFieldSelection {
+  node: FieldSelection;
+}
+
+export function isFieldSelection(selection: FieldSelectionType): selection is FieldSelection {
+  return !!(selection && typeof selection == "object" && !("edges" in selection));
```

Comment:
Will this cause us to break `compileSelect` if they add a model field called `edges`?

Reply:
Which we already have 2 models in prod which would have hit this case.

Reply:
Good catch, this broke a specific case when a `BelongsTo` has a node called edges. I have added additional check that also tests that if there is a field called `edges` in the selection that it is also an object (indicating that it is selecting a connection). I have also added a bunch of additional tests for each of the cases.